src/main/scala/io/iohk/ethereum/db/cache/MapCache.scala

```diff
@@ -1,53 +1,56 @@
 package io.iohk.ethereum.db.cache
 
 import java.util.concurrent.TimeUnit
-import java.util.concurrent.atomic.AtomicLong
+import java.util.concurrent.locks.ReentrantReadWriteLock
 
 import io.iohk.ethereum.utils.Config.NodeCacheConfig
 
 import scala.collection.{Seq, mutable}
 import scala.concurrent.duration.FiniteDuration
 
 //TODO EC-492 Investigate more carefully possibility of having read cache in front of db
-// class is not entirely thread safe
-// All updates need to be atomic and visible in respect to get, as get may be called from other threads.
-// Other methods are only called from actor context, and all updates are always visible to them
+// class is now entirely thread safe
```

Comment:
i think this comment can be removed, and make it explicit at code by renaming the class to: `ThreadSafeMapCache`

Reply:
Hmm.. Cache is something that is shared... I believe its thread safety should be assumed by default. Maybe we shouldn't rename it.